pkg/cmd/pulumi/new.go

```diff
@@ -658,6 +711,29 @@ func promptAndCreateStack(ctx context.Context, b backend.Backend, prompt promptF
 			return nil, err
 		}
 		s, err := stackInit(ctx, b, formattedStackName, root, setCurrent, secretsProvider)
+		// Check if stack already exists and allow the user to bypass it.
+		var e *backend.StackAlreadyExistsError
+		if errors.As(err, &e) {
+			accept, err := acceptStackAlreadyExists(stackName, opts)
+			if err != nil {
+				return nil, err
+			}
+			if !accept {
+				continue
+			}
+			stackRef, err := b.ParseStackReference(stackName)
+			if err != nil {
+				return nil, err
+			}
+			s, err = b.GetStack(ctx, stackRef)
+			if err != nil {
+				return nil, err
+			}
+			if err = state.SetCurrentStack(stackRef.String()); err != nil {
```

Comment:
Shouldn't this still respect `setCurrent`?

Reply:
Can you elaborate?

Reply:
Not allowing `pulumi new` to use an existing stack anymore.